### PR TITLE
[DEVOPS-107] Fix CI

### PR DESCRIPTION
**Pull request by domenkozar**
Superseeds https://github.com/input-output-hk/cardano-sl/pull/346

### CODE REVIEW

**Comment by domenkozar** [comment]
```
May 30 13:27:35 [250 of 265] Compiling Pos.Launcher.Runner ( src/Pos/Launcher/Runner.hs, .stack-work/dist/x86_64-osx-nix/Cabal-1.24.2.0/build/Pos/Launcher/Runner.o )
May 30 13:31:57 [251 of 265] Compiling Pos.Launcher.Scenario ( src/Pos/Launcher/Scenario.hs, .stack-work/dist/x86_64-osx-nix/Cabal-1.24.2.0/build/Pos/Launcher/Scenario.o )
May 30 13:39:53 [252 of 265] Compiling Pos.Launcher.Launcher ( src/Pos/Launcher/Launcher.hs, .stack-work/dist/x86_64-osx-nix/Cabal-1.24.2.0/build/Pos/Launcher/Launcher.o )
May 30 13:47:38 [253 of 265] Compiling Pos.Launcher     ( src/Pos/Launcher.hs, .stack-work/dist/x86_64-osx-nix/Cabal-1.24.2.0/build/Pos/Launcher.o )
May 30 13:47:47 [254 of 265] Compiling Pos.Wallet.Launcher.Param ( src/Pos/Wallet/Launcher/Param.hs, .stack-work/dist/x86_64-osx-nix/Cabal-1.24.2.0/build/Pos/Wallet/Launcher/Param.o )
```

Launcher part of compilation takes 20min to build

File: util-scripts/travis.sh [hunk 1]
```diff
@@ -29,7 +29,7 @@ stack --nix --no-terminal --local-bin-path daedalus/ install cardano-sl \
 if [[ "$TRAVIS_OS_NAME" == "linux" && "$TRAVIS_BRANCH" == "master" && "$TRAVIS_PULL_REQUEST" == "false" ]]; then
   ./update_wallet_web_api_docs.sh
   ./update_cli_docs.sh
-#  ./update_haddock.sh
+  ./update_haddock.sh
```
Comment (int-index) [inline util-scripts/travis.sh]: What about uncommenting `export EXTRA_STACK="--haddock"` above?

Comment (domenkozar) [inline util-scripts/travis.sh]: Good point, pushed a fix.

**Comment by gromakovsky** [comment]
Now haddock fails with scary error :(
http://i.imgur.com/orIQ7EQ.png

**Comment by gromakovsky** [comment]
Oh, it's not scary.

**Comment by gromakovsky** [comment]
@neongreen probably can fix it.

**Comment by neongreen** [comment]
Okay, let's see whether it works or not

**Comment by neongreen** [comment]
It works but now it says `No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.` despite output seemingly being printed. @domenkozar any ideas how to fix it?

**Comment by domenkozar** [comment]
@neongreen pushed a fix to wait 20min on build output.

**Comment by domenkozar** [comment]
@neongreen @nh2 @int-index haddock still runs out of memory on linux:

```
Jun 06 12:18:20     Process exited with code: ExitFailure (-9) (THIS MAY INDICATE OUT OF MEMORY)
```

https://travis-ci.org/input-output-hk/cardano-sl/jobs/239926209#L3480

**Comment by nh2** [comment]
@domenkozar That's an interesting find that haddock fails on this module, maybe it'll help us to learn some more insights on this issue!

**Comment by domenkozar** [comment]
@nh2 I'm going to merge this to fix travis-ci to include the primary job. But haddock is still disabled :)

**Pull request merged**